bluemira/radiation_transport/radiation_profile.py

```diff
@@ -0,0 +1,2151 @@
+        self.imp_data_core = impurity_data_core
+        self.imp_content_sol = impurity_content_sol
+        self.imp_data_sol = impurity_data_sol
+        self.lcfs = self.eq.get_LCFS()
+
+        # Midplane parameters
+        self.psi_n = psi_n
+        self.ne_mp = ne_mp
+        self.te_mp = te_mp
+
+        # To be calculated calling analyse
+        self.core_rad = None
+        self.sol_rad = None
+
+        # To be calculated calling rad_map
+        self.x_tot = None
+        self.z_tot = None
+        self.rad_tot = None
+
+    def analyse(self, firstwall_geom: Grid):
+        """
+        Using core radiation model and sol radiation model
+        to calculate the radiation source at all points
+
+        Parameters
+        ----------
+        first_wall: Loop
+            The closed first wall geometry
+
+        Returns
+        -------
+        x_all: np.array
+            The x coordinates of all the points included within the flux surfaces
+        z_all: np.array
+            The z coordinates of all the points included within the flux surfaces
+        rad_all: np.array
+            The local radiation source at all points included within
+            the flux surfaces [MW/m^3]
+        """
+        self.core_rad = CoreRadiation(
+            self.eq,
+            self.flux_surf_solver,
+            self.params,
+            self.psi_n,
+            self.ne_mp,
+            self.te_mp,
+            self.imp_content_core,
+            self.imp_data_core,
+        )
+
+        if self.eq.is_double_null:
+            self.sol_rad = DNScrapeOffLayerRadiation(
+                self.eq,
+                self.flux_surf_solver,
+                self.params,
+                self.imp_content_sol,
+                self.imp_data_sol,
+                firstwall_geom,
+            )
+        else:
+            self.sol_rad = SNScrapeOffLayerRadiation(
+                self.eq,
+                self.flux_surf_solver,
+                self.params,
+                self.imp_content_sol,
+                self.imp_data_sol,
+                firstwall_geom,
+            )
+
+        return self.core_rad, self.sol_rad
+
+    def rad_core_by_psi_n(self, psi_n):
+        """
+        Calculation of core radiation source for a given (set of) psi norm value(s)
+
+        Parameters
+        ----------
+        psi_n: float (list)
+            The normalised magnetic flux value(s)
+
+        Returns
+        -------
+        rad_new: float (list)
+            Local radiation source value(s) associated to the given psi_n
+        """
+        core_rad = CoreRadiation(
+            self.eq,
+            self.flux_surf_solver,
+            self.params,
+            self.psi_n,
+            self.ne_mp,
+            self.te_mp,
+            self.imp_content_core,
+            self.imp_data_core,
+        )
+        core_rad.build_mp_rad_profile()
+        rad_tot = np.sum(np.array(core_rad.rad, dtype=object), axis=0).tolist()
```

Comment:
I doubt you need to convert it to a list (you might need to call `.ravel()` if it needs to be 1d)
```suggestion
        rad_tot = np.sum(np.array(core_rad.rad, dtype=object), axis=0)
```